polaris/viz/spherical.py

```diff
@@ -71,59 +64,52 @@ def plot_global_mpas_field(mesh_filename, da, out_filename, config,
     patch_edge_color : str, optional
         The color of patch edges (if not the same as the face)
     """
-    matplotlib.use('agg')
-    uxgrid = ux.open_grid(mesh_filename)
-    uxda = ux.UxDataArray(da, uxgrid=uxgrid)
-
-    gdf_data = uxda.to_geodataframe()
 
-    hv.extension('matplotlib')
+    use_mplstyle()
 
+    transform = cartopy.crs.Geodetic()
     projection = cartopy.crs.PlateCarree(central_longitude=central_longitude)
 
-    colormap = config.get(colormap_section, 'colormap_name')
-    cmap = cm.get_cmap(colormap)
-    if config.has_option(colormap_section, 'under_color'):
-        under_color = config.get(colormap_section, 'under_color')
-        cmap.set_under(under_color)
-    if config.has_option(colormap_section, 'over_color'):
-        over_color = config.get(colormap_section, 'over_color')
-        cmap.set_over(over_color)
+    mesh_ds = xr.open_dataset(mesh_filename)
+    descriptor = mosaic.Descriptor(
+        mesh_ds, projection=projection, transform=transform, use_latlon=True
+    )
 
-    norm_type = config.get(colormap_section, 'norm_type')
-    if norm_type == 'linear':
-        logz = False
-    elif norm_type == 'log':
-        logz = True
-    else:
-        raise ValueError(f'Unsupported norm_type: {norm_type}')
+    fig, ax = plt.subplots(figsize=figsize,
+                           constrained_layout=True,
+                           subplot_kw=dict(projection=projection))
 
-    colorbar_limits = config.getlist(colormap_section, 'colorbar_limits',
-                                     dtype=float)
+    if title is not None:
+        fig.suptitle(title, y=0.935)
 
-    plot = gdf_data.hvplot.polygons(
-        c=da.name, cmap=cmap, logz=logz,
-        clim=tuple(colorbar_limits),
-        clabel=colorbar_label,
-        width=1600, height=800, title=title,
-        xlabel="Longitude", ylabel="Latitude",
-        projection=projection,
-        rasterize=True)
+    colormap, norm, ticks = _setup_colormap(config, colormap_section)
+
+    pcolor_kwargs = dict(
+        cmap=colormap, norm=norm, zorder=1, edgecolors='face', linewidths=0.2
+    )
+
+    if patch_edge_color is not None:
+        pcolor_kwargs['edgecolors'] = patch_edge_color
+
+    gl = ax.gridlines(
+        color='gray', linestyle=':', zorder=5, draw_labels=True, linewidth=0.5
+    )
+    gl.right_labels = False
+    gl.top_labels = False
+
+    pc = mosaic.polypcolor(ax, descriptor, da, **pcolor_kwargs)
 
     if plot_land:
-        plot = plot * gf.land * gf.coastline
+        ax._add_land_lakes_coastline(ax)
```

Comment:
It's generally discouraged to call protected methods like this.  Do you know if this is an exception, and if so, why is it protected?

Reply:
Ah, and unfortunately it never got tested because we don't currently have any realistic tests where plotting continents makes sens.